src/server/replica.cc

```diff
@@ -467,9 +467,11 @@ error_code Replica::InitiateDflySync() {
   RETURN_ON_ERR(cntx_.SwitchErrorHandler(std::move(err_handler)));
 
   // Make sure we're in LOADING state.
-  // TODO: Flush db on retry.
   CHECK(service_.SwitchState(GlobalState::ACTIVE, GlobalState::LOADING) == GlobalState::LOADING);
 
+  // Flush dbs.
+  JournalExecutor{&service_}.FlushAll();
+
```

Comment:
Not sure about the order here as well - but it probably doesn't matter.

The journal executor can execute commands in loading state because of its special flag